File: loja-java/src/main/java/loja/Produto.java
package loja;

public class Produto {
    private final String nome;
    private final double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public String nome() {
        return nome;
    }

    public double preco() {
        return preco;
    }
}

File: loja-ruby/lib/loja/produto.rb
module Loja
  class Produto
    attr_reader :nome, :preco

    def initialize(nome, preco)
      @nome = nome
      @preco = preco
    end

    def to_s
      "#{@nome} - R$ #{format('%.2f', @preco)}"
    end
  end
end
